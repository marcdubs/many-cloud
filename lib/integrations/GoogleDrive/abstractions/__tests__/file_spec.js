const File = require("../file");

describe("file", () => {
  describe("constructor", () => {
    const expected_id = "1234",
      expected_connection = "the_connection",
      expected_type = "file";
    let actual;
    beforeEach(() => {
      actual = new File(expected_id, expected_connection);
    });

    it("sets the id", () => {
      expect(actual.id).toEqual(expected_id);
    });

    it("sets the connection", () => {
      expect(actual.connection).toEqual(expected_connection);
    });

    it("sets the type", () => {
      expect(actual.type).toEqual(expected_type);
    });
  });

  describe("download_to", () => {
    describe("when successfull", () => {
      let file, connection;
      beforeEach(async done => {
        connection = {
          download_file: jest.fn().mockImplementation(() => {
            return new Promise(resolve => {
              resolve();
            });
          })
        };
        file = new File("1234", connection);
        await file.download_to("destination");
        done();
      });

      it("calls connection.download_file", () => {
        expect(connection.download_file).toHaveBeenCalled();
      });
    });

    describe("when unsuccessfull", () => {
      const expected_error = "Well'a look what you did done did!";
      let file, connection;
      beforeEach(() => {
        connection = {
          download_file: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              reject(new Error(expected_error));
            });
          })
        };
        file = new File("1234", connection);
      });

      it("rejects with the error", async () => {
        await expect(
          file.download_to("destination")
        ).rejects.toThrowErrorMatchingSnapshot();
      });
    });
  });
});
