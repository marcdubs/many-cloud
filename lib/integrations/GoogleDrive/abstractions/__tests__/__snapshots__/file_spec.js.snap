// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file download_to when unsuccessfull rejects with the error 1`] = `"Well'a look what you did done did!"`;
